lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -501,7 +507,7 @@ static GraphOperationInst *simplifyOperands(GraphOperationInst *origInst,
     auto storeOwnership = hasOwnership ? StoreOwnershipQualifier::Trivial
                                        : StoreOwnershipQualifier::Unqualified;
     assert(newInst->getNumResults() == 1 &&
-           "simplified graph_op should still only have one result");
+           "out parameter should turn into 1 result");
```

Comment:
1. if a generic graph_op produces a list of outputs (say infeed dequeue), and it's produced by silgen, will its output param contain > 1 result?

2. Also, should we move this check to (or replicate it in) an earlier stage (like the body of "if (argumentLowering == GraphOperationInfo::ArgumentLowering::Out)" above), which might facilitate debugging in case it's violated?

Reply:
update on question `#1`: i guess in that case it'll still be in the aggregate form. maybe elaborate the comment as such?

Reply:
I realized that I can simplify the code by moving this store stuff into the branch where the `newInst` is constructed. This lets me remove the assert because the construction explicitly constructs it with 1 output result.

To your point 2, there is an earlier assert `assert(!outParameterAddress && "there is more than one out parameter");` that does what you are asking. I added a comment to that explaining that the output is still a single aggregate.